pandas/core/arrays/_arrow_utils.py

```diff
@@ -25,8 +27,16 @@ def pyarrow_array_to_numpy_and_mask(arr, dtype):
         Tuple of two numpy arrays with the raw data (with specified dtype) and
         a boolean mask (validity mask, so False means missing)
     """
+    dtype = np.dtype(dtype)
```

Comment:
This should not be needed, as we can ensure internally to always pass a numpy dtype object. I see there is one occurrence in `PeriodDtype` where we pass a string type -> you can update that case to pass `np.dtype("int64")` instead.

Reply:
Yes, that was exactly the reason I added the line. I changed `PeriodDtype` the way you suggested.

Unfortunately some other tests break when I remove the line. IIUC pyarrow is using the function as well.

```shell
______________________________________________ test_arrow_sliced _____________________________________________________________________________________

    @td.skip_if_no("pyarrow", min_version="0.16.0")
    def test_arrow_sliced():
        # https://github.com/pandas-dev/pandas/issues/38525
        import pyarrow as pa
    
        df = pd.DataFrame({"a": pd.array([0, None, 2, 3, None], dtype="Int64")})
        table = pa.table(df)
>       result = table.slice(2, None).to_pandas()

pandas/tests/arrays/masked/test_arrow_compat.py:65: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
pyarrow/array.pxi:751: in pyarrow.lib._PandasConvertible.to_pandas
    ???
pyarrow/table.pxi:1668: in pyarrow.lib.Table._to_pandas
    ???
/opt/conda/lib/python3.8/site-packages/pyarrow/pandas_compat.py:792: in table_to_blockmanager
    blocks = _table_to_blocks(options, table, categories, ext_columns_dtypes)
/opt/conda/lib/python3.8/site-packages/pyarrow/pandas_compat.py:1133: in _table_to_blocks
    return [_reconstruct_block(item, columns, extension_columns)
/opt/conda/lib/python3.8/site-packages/pyarrow/pandas_compat.py:1133: in <listcomp>
    return [_reconstruct_block(item, columns, extension_columns)
/opt/conda/lib/python3.8/site-packages/pyarrow/pandas_compat.py:751: in _reconstruct_block
    pd_ext_arr = pandas_dtype.__from_arrow__(arr)
pandas/core/arrays/numeric.py:65: in __from_arrow__
    data, mask = pyarrow_array_to_numpy_and_mask(arr, dtype=self.type)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

arr = <pyarrow.lib.Int64Array object at 0x7f77e8678820>
[
  2,
  3,
  null
], dtype = <class 'numpy.int64'>

    def pyarrow_array_to_numpy_and_mask(arr, dtype):
        """
        Convert a primitive pyarrow.Array to a numpy array and boolean mask based
        on the buffers of the Array.
    
        At the moment pyarrow.BooleanArray is not supported.
    
        Parameters
        ----------
        arr : pyarrow.Array
        dtype : numpy.dtype
    
        Returns
        -------
        (data, mask)
            Tuple of two numpy arrays with the raw data (with specified dtype) and
            a boolean mask (validity mask, so False means missing)
        """
        # dtype = np.dtype(dtype)
        buflist = arr.buffers()
        # Since Arrow buffers might contain padding and the data might be offset,
        # the buffer gets sliced here before handing it to numpy.
        # See also https://github.com/pandas-dev/pandas/issues/40896
>       offset = arr.offset * dtype.itemsize
E       TypeError: unsupported operand type(s) for *: 'int' and 'getset_descriptor'

pandas/core/arrays/_arrow_utils.py:35: TypeError
```

Reply:
> This should not be needed, as we can ensure internally to always pass a numpy dtype object. I see there is one occurrence in `PeriodDtype` where we pass a string type -> you can update that case to pass `np.dtype("int64")` instead.

The change form `"int64"` to `np.dtype("int64")` caused some trouble with timezone conversion on Windows. Revertig it lets the tests pass.
```shell
Traceback (most recent call last):
  File "pandas\_libs\tslibs\tzconversion.pyx", line 385, in pandas._libs.tslibs.tzconversion.tz_convert_utc_to_tzlocal
    return _tz_convert_tzlocal_utc(utc_val, tz, to_utc=False, fold=fold)
  File "pandas\_libs\tslibs\tzconversion.pyx", line 596, in pandas._libs.tslibs.tzconversion._tz_convert_tzlocal_utc
    delta = _tzlocal_get_offset_components(val, tz, to_utc, fold)
  File "pandas\_libs\tslibs\tzconversion.pyx", line 556, in pandas._libs.tslibs.tzconversion._tzlocal_get_offset_components
    dt = dt.astimezone(tz)
  File "C:\Miniconda\envs\pandas-dev\lib\site-packages\dateutil\tz\_common.py", line 144, in fromutc
    return f(self, dt)
  File "C:\Miniconda\envs\pandas-dev\lib\site-packages\dateutil\tz\_common.py", line 261, in fromutc
    _fold = self._fold_status(dt, dt_wall)
  File "C:\Miniconda\envs\pandas-dev\lib\site-packages\dateutil\tz\_common.py", line 196, in _fold_status
    if self.is_ambiguous(dt_wall):
  File "C:\Miniconda\envs\pandas-dev\lib\site-packages\dateutil\tz\tz.py", line 254, in is_ambiguous
    naive_dst = self._naive_is_dst(dt)
  File "C:\Miniconda\envs\pandas-dev\lib\site-packages\dateutil\tz\tz.py", line 260, in _naive_is_dst
    return time.localtime(timestamp + time.timezone).tm_isdst
OSError: [Errno 22] Invalid argument
```

Reply:
OK, that's a good reason to keep it then!